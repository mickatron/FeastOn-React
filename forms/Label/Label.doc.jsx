/* eslint-disable */
import React from 'react';
import Component from './Label';

/**
 * A basic tile device often used to group content or implement hierarchy.
 * 
 * ** Sub Components **
 *
 * @composedWith UiWrapper
*/
function Label(props) {
  return <Component {...props} />;
}

export default Label;
