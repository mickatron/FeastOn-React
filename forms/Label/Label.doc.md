

### Default Label.
```js
<Label  
  text="Label text"
  input={({uid}) => (
    <input id={uid} />
  )}
/>

```