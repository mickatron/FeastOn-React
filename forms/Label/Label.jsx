import PropTypes from 'prop-types';
import React from 'react';

import uniqueId from 'lodash/uniqueId';

import uiWrapper from '../../HOC/uiWrapper';


class Label extends React.PureComponent {
  static propTypes = {
    /** The Label components child elements. */
    text: PropTypes.node.isRequired,
    /** A Render Prop callback used to render the input element associated to the label.
     * If we used `this.props.children` instead the elements `id` prop would have to conform
     * to some interface.
     * Using a render prop we can pass in custom and third-party input Components. */
    input: PropTypes.func.isRequired,
    /** Should the label be rendered to the left or right of the input. */
    renderInput: PropTypes.oneOf(['left', 'right']),
  }

  static defaultProps = {
    renderInput: 'right',
  }

  componentWillMount() {
    // generate a unique ID for the component
    // TODO: test and update for isomorphic applications
    this.uid = uniqueId();
  }

  uid = undefined;

  render() {
    const {
      text,
      input,
      renderInput,
    } = this.props;

    const { uid } = this;
    return (
      <React.Fragment>
        {renderInput === 'left' && input({ uid })}
        <span className="f-Label__text">{text}</span>
        {renderInput === 'right' && input({ uid })}
      </React.Fragment>
    );
  }
}

export default uiWrapper({
  name: 'Label',
  Component: Label,
  element: 'label',
});
