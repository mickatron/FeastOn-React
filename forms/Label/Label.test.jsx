import { shallow } from 'enzyme';
import React from 'react';
import Label from './Label';

describe('Label', () => {
  let component = null;

  it('renders', () => {
    component = shallow((
      <Label
        text="Label text"
        input={({ uid }) => (
          <input id={uid} />
        )}
      />
    ));
    expect(component).toMatchSnapshot();
  });
});
