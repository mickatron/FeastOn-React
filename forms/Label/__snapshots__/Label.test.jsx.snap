// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UiWrapper
    as="div"
    className={undefined}
    input={[Function]}
    modifier={undefined}
    text="Label text"
    widthClass={undefined}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Label
        input={[Function]}
        renderInput="right"
        text="Label text"
      />,
      "className": "Label",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "input": [Function],
        "renderInput": "right",
        "text": "Label text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Label
          input={[Function]}
          renderInput="right"
          text="Label text"
        />,
        "className": "Label",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "input": [Function],
          "renderInput": "right",
          "text": "Label text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
